crates/polars-core/src/datatypes/dtype.rs

```diff
@@ -789,3 +789,26 @@ pub fn create_enum_data_type(categories: Utf8ViewArray) -> DataType {
     let rev_map = RevMapping::build_local(categories);
     DataType::Enum(Some(Arc::new(rev_map)), Default::default())
 }
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct PlFlavor {
```

Comment:
Can we call this `CompatLevel` now we expose this to users?

Reply:
Do you mean change all argument names in existing code? `pl_flavor`, as an argument name, exists in many function signatures. It was originally `pl_flavor: bool` and are now `pl_flavor: PlFlavor`. I think it looks noisy to have `pl_flavor: CompatLevel` in function signatures.

Reply:
We can name that `level: CompatLevel`.

Reply:
I think it still sounds strange. `CompatLevel::highest()` means least compatible, which is counter-intuitive. Maybe we can name it `FutureLevel` or so. What's your opinion?

Reply:
The thing is, is that we are now functionality intended to be public, where `flavor` never was meant to be public.

So what I want in the public signature is `compat_level: CompatLevel`. The python `future` argument can map to that new `compat_level`. And then later on 2.0, we can remove `future`.

I don't like `Future` as a public name for this. IMO `compat_level` is better as that describes what it influences. E.g. how compatible is the output arrow data.

To deal with the confusion, we can call it `CompatLevel::newest` instead of `highest`.